src/util/misc.py

```diff
@@ -380,59 +380,80 @@ def __init__(self, name):
     def __getattr__(self, attr):
         return SelfPrint("%r.%s" % (self, attr))
 
+    def __getitem__(self, key):
+        return SelfPrint("%r[%r]" % (self, key))
+
     def __call__(self, *args, **kwargs):
         s = [repr(a) for a in args]
         for k, v in kwargs.items():
             s.append("%s=%r" % (k, v))
         return SelfPrint("%r(%s)" % (self, ", ".join(s)))
 
     def __add__(self, other):
-        return SelfPrint("%r + %r" % (self, other))
+        return SelfPrint("(%r + %r)" % (self, other))
 
     def __radd__(self, other):
-        return SelfPrint("%r + %r" % (other, self))
+        return SelfPrint("(%r + %r)" % (other, self))
 
     def __sub__(self, other):
-        return SelfPrint("%r - %r" % (self, other))
+        return SelfPrint("(%r - %r)" % (self, other))
 
     def __rsub__(self, other):
-        return SelfPrint("%r - %r" % (other, self))
+        return SelfPrint("(%r - %r)" % (other, self))
 
     def __mul__(self, other):
-        return SelfPrint("%r * %r" % (self, other))
+        return SelfPrint("(%r * %r)" % (self, other))
 
     def __rmul__(self, other):
-        return SelfPrint("%r * %r" % (other, self))
+        return SelfPrint("(%r * %r)" % (other, self))
 
     def __div__(self, other):
-        return SelfPrint("%r / %r" % (self, other))
+        return SelfPrint("(%r / %r)" % (self, other))
 
     def __rdiv__(self, other):
-        return SelfPrint("%r / %r" % (other, self))
+        return SelfPrint("(%r / %r)" % (other, self))
 
     def __truediv__(self, other):
-        return SelfPrint("%r / %r" % (self, other))
+        return SelfPrint("(%r / %r)" % (self, other))
 
     def __rtruediv__(self, other):
-        return SelfPrint("%r / %r" % (other, self))
+        return SelfPrint("(%r / %r)" % (other, self))
 
     def __floordiv__(self, other):
-        return SelfPrint("%r // %r" % (self, other))
+        return SelfPrint("(%r // %r)" % (self, other))
 
     def __rfloordiv__(self, other):
-        return SelfPrint("%r // %r" % (other, self))
+        return SelfPrint("(%r // %r)" % (other, self))
 
     def __mod__(self, other):
-        return SelfPrint("%r %% %r" % (self, other))
+        return SelfPrint("(%r %% %r)" % (self, other))
 
     def __rmod__(self, other):
-        return SelfPrint("%r %% %r" % (other, self))
+        return SelfPrint("(%r %% %r)" % (other, self))
+
+    def __pow__(self, other):
+        return SelfPrint("(%r ** %r)" % (self, other))
+
+    def __rpow__(self, other):
+        return SelfPrint("(%r ** %r)" % (other, self))
+
+    def __pos__(self):
+        return SelfPrint("+(%r)" % (self,))
+
+    def __neg__(self):
+        return SelfPrint("-(%r)" % (self,))
 
     def __repr__(self):
         return self.__name
 
     __str__ = __repr__
 
+    def __iter__(self):
+        raise RuntimeError
```

Comment:
```suggestion
        raise RuntimeError("Cannot self-print iteration")
```
I'd rather make the error less mysterious
```
  File "/home/odoo/src/odoo/16.0/odoo/tools/safe_eval.py", line 405, in safe_eval
    raise ValueError('%s: "%s" while evaluating\n%r' % (ustr(type(e)), ustr(e), expr))
ValueError: <class 'RuntimeError'>: "" while evaluating
```